### PR TITLE
Feature: DynamoDB DQL query runner

### PR DESCRIPTION
**Pull request by alexanderlz**
Adds the ability to query DynamoDB using SQL!
(with the help of the following library 
https://github.com/mathcamp/dql
)

### CODE REVIEW

File: redash/utils/__init__.py [hunk 1]
```diff
@@ -66,6 +66,9 @@ def default(self, o):
         if isinstance(o, datetime.timedelta):
             return str(o)
 
+        if isinstance(o, unicode):
```
Comment (arikfr) [inline redash/utils/__init__.py]: Why is this needed? :open_mouth:

File: redash/query_runner/dynamodb_sql.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                    "type": "string",
+                    "default": "anything"
+
+                },
+                "is_secure": {
+                    "type": "boolean",
+                    "default": False,
+                }
+            },
+            "required": ["host"]
```
Comment (arikfr) [inline redash/query_runner/dynamodb_sql.py]: Please add a `secret` field with array of the secrets (in this case `SECRET_KEY`).

File: requirements.txt [hunk 3]
```diff
@@ -19,7 +19,8 @@ python-dateutil==2.4.2
 pytz==2015.7
 redis==2.7.5
 requests==2.3.0
-six==1.5.2
+botocore==1.4.4
```
Comment (arikfr) [inline requirements.txt]: ?

File: redash/query_runner/dynamodb_sql.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            raise sys.exc_info()[1], None, sys.exc_info()[2]
+
+    def run_query(self, query):
+
+        connection = None
+        try:
+            engine = FragmentEngine()
+            connection = engine.connect(**self.configuration.to_dict())
+
+            resDict = engine.execute(query if str(query).endswith(';') else str(query)+';')
```
Comment (arikfr) [inline redash/query_runner/dynamodb_sql.py]: `res_dict`...

File: redash/query_runner/dynamodb_sql.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            "secret": ["secret_key"]
+        }
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+    @classmethod
+    def type(cls):
+        return "ddb_sql"
```
Comment (arikfr) [inline redash/query_runner/dynamodb_sql.py]: `dynamodb_dql`

Comment (arikfr) [inline redash/query_runner/dynamodb_sql.py]: Also please add a `name` method that will return `DynamoDB (with DQL)`.

File: requirements.txt (hunk 3 above)
Comment (alexanderlz) [inline requirements.txt]: had to update the six lib (to 1.10.0), to be compatible with dql

**Comment by arikfr** [comment]
Thanks!

**Comment by mvolpato** [comment]
Can you please give an explanation of the fields used by DynamoDB connection? Some are clear to me (Access_key, Secret_key, Region), the others are not.

I get this error:
```
Error running query: InvalidSignatureException: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details. The Canonical String for this request should have been 'POST / content-type:application/x-amz-json-1.0 host:dynamodb.eu-west-1.amazonaws.com user-agent:Botocore/1.4.4 Python/2.7.11 Linux/3.13.0-85-generic x-amz-date:20160516T194838Z x-amz-target:DynamoDB_20120810.DescribeTable content-type;host;user-agent;x-amz-date;x-amz-target [hash here]' The String-to-Sign should have been 'AWS4-HMAC-SHA256 20160516T194838Z 20160516/eu-west-1/dynamodb/aws4_request [hash here]' Args: {'TableName': 'tablename'}
```

I used `dynamodb.eu-west-1.amazonaws.com` as Host, is_secore false, Port 80 (or 8000), Region eu-west-1.

**Comment by alexanderlz** [comment]
Hi @mvolpato,
Which boto3 version are you using?

**Comment by mvolpato** [comment]
Hi @alexanderlz 
No idea. I simply used the container provided by Bitnami on AWS.

**Comment by arikfr** [comment]
@mvolpato see #1059. It's a bug. Once #1059 is merged, just create a new DynamoDB data source without specifying host.

**Comment by mvolpato** [comment]
Thank you @arikfr. Looking forward for the release.
